test/components/patch_serviceaccount_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build aks aws aws_edge packet
+// +build e2e
+
+// nolint:testpackage
+package components
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+
+	_ "github.com/kinvolk/lokomotive/pkg/components/flatcar-linux-update-operator"
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	retryInterval  = time.Second * 5
+	timeout        = time.Minute * 5
+	contextTimeout = 10
+)
+
+type componentTestCase struct {
+	namespace string
+}
+
+func TestDisableAutomountServiceAccountToken(t *testing.T) {
+	client := testutil.CreateKubeClient(t)
+
+	ns, err := client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatal("finding namespaces: %w", err)
+	}
+
+	var componentTestCases []componentTestCase
+
+	for _, val := range ns.Items {
+		if !isSystemNamespace(val.Name) {
+			componentTestCases = append(componentTestCases, componentTestCase{
+				namespace: val.Name,
+			})
+		}
+	}
+
+	for _, tc := range componentTestCases {
+		tc := tc
+		t.Run(tc.namespace, func(t *testing.T) {
+			t.Parallel()
+
+			platform := os.Getenv("PLATFORM")
+			if platform == "" {
+				t.Fatal("env var PLATFORM was not set")
+			}
+
+			if testutil.Platform(platform) != testutil.PlatformAKS {
```

Comment:
Why don't we test against the AKS? If we don't want to do that, we can remove `aks` from builds tags.

We also have `util.IsPlatformSupported()` function, which might be helpful.